kotlin-plugin/src/main/kotlin/dev/nemecec/kotlinlogging/compiletimeplugin/KotlinLoggingIrGenerationExtension.kt

```diff
@@ -543,8 +543,13 @@ class KotlinLoggingIrGenerationExtension(
                               endOffset = UNDEFINED_OFFSET,
                               symbol = extensionReceiverValueParameter!!.symbol,
                             )
-                          parent = currentDeclarationParent!!
-                          setRegularArgument(0, loggingCallExpressions.messageExpression)
+                          parent = this@createLambdaIrSimpleFunction
```

Comment:
`parent` in these calls is resolved to `this@createLambdaIrSimpleFunction.parent`, because `IrBlockBody` does not have a `parent` property. Therefore, you create a recursive dependency here:
```
this@createLambdaIrSimpleFunction.parent = this@createLambdaIrSimpleFunction
```

This hangs the compiler

Reply:
@AbdullinAM, I now applied a fix. Have a look, does it look reasonable? :-)